kubernetes.go

```diff
@@ -0,0 +1,90 @@
+package veneur
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type KubernetesDiscoverer struct {
+	clientset *kubernetes.Clientset
+}
+
+func NewKubernetesDiscoverer() (*KubernetesDiscoverer, error) {
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return &KubernetesDiscoverer{clientset}, nil
+}
+
+func (kd *KubernetesDiscoverer) GetDestinationsForService(serviceName string) ([]string, error) {
+	pods, err := kd.clientset.CoreV1().Pods(metav1.NamespaceAll).List(metav1.ListOptions{
+		LabelSelector: "app=veneur-global",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	ips := make([]string, 0, len(pods.Items))
+	for podIndex, pod := range pods.Items {
+
+		var forwardPort string
+
+		if pod.Status.Phase != v1.PodRunning {
+			continue
+		}
+
+		// TODO don't assume there is only one container for the veneur global
```

Comment:
Is this todo relevant?

Reply:
Yep. In theory, we could be running two containers in the same pod for veneur-global. If so, that'll break this logic.